src/main/java/vct/col/util/AbstractTypeCheck.java

```diff
@@ -1706,8 +1706,12 @@ private void force_frac(ASTNode arg) {
       Warning("Encountered an integer division ('/') '%s' where a fraction was expected, did you mean a fraction division ('\\') here?", arg);
     }
 
-    if(arg.getType().isPrimitive(PrimitiveSort.Integer)) {
-      arg.setType(new PrimitiveType(PrimitiveSort.Fraction));
+    if (arg instanceof ConstantExpression && arg.getType().isPrimitive(PrimitiveSort.Integer)) {
+      if (arg.equals(0)) {
+        arg.setType(new PrimitiveType(PrimitiveSort.ZFraction));
```

Comment:
We change the type of the argument. Why is that done in TypeCheck? To me "check" signals that it does not have any side effects however since we change the type it does have side effects.

Why do not have to change the value of the argument?

Reply:
Indeed typechecking should be side-effect free, but currently typechecking does have some side effects. I think we will be able to fix that in the new AST, but for now I think we can keep it as is in the old structure.